Makefile

```diff
@@ -68,7 +68,7 @@ bench: ## runs benchmark tests
 	rm -f pipe output.txt mem.prof cpu.prof bench.test
 	mkfifo pipe
 	tee output.txt < pipe &
-	go test -tags bench -benchmem -bench=. ./test/bench -memprofile=mem.prof -cpuprofile=cpu.prof > pipe
+	go test -tags bench -benchmem -bench=. ./test/bench -memprofile=mem.prof -cpuprofile=cpu.prof -timeout=20m > pipe
```

Comment:
The CI execution time has slowed down due to benchmark tests, posing a risk of slowing down the development cycle. It would be ideal if the CI could be completed within 15 minutes.

How about removing the big tests if it could show that the result tends to be slower in linear?